src/os/chain_xattr.cc

```diff
@@ -135,25 +136,32 @@ int chain_getxattr(const char *fn, const char *name, void *val, size_t size)
   do {
     chunk_size = (size < CHAIN_XATTR_MAX_BLOCK_LEN ? size : CHAIN_XATTR_MAX_BLOCK_LEN);
     get_raw_xattr_name(name, i, raw_name, sizeof(raw_name));
-    size -= chunk_size;
 
     r = sys_getxattr(fn, raw_name, (char *)val + pos, chunk_size);
+    if (i && r == -ENODATA) {
```

Comment:
ENODATA may be ENOATTR better

BSD, Linux, DARWIN all with return ENOATTR and ENOATTR is defined to be ENODATA  in Linux